Change all julialang.org links to https

tkelman opened the pull request:
The first commit here was from running `contrib/add_license_to_files.jl` then
```
sed -i 's|http://julialang.org|https://julialang.org|g' $(git grep -l http://julialang.org)
sed -i 's|http://docs.julialang.org|https://docs.julialang.org|g' $(git grep -l http://docs.julialang.org)
```

tkelman committed:
Change all julialang.org links to https

omus commented on CONTRIBUTING.md:
```diff
@@ -1,18 +1,18 @@
 ## Learning Julia
 
-[The learning page](http://julialang.org/learning/) has a great list of resources for new and experienced users alike. [This tutorial video](https://www.youtube.com/watch?v=vWkgEddb4-A) is one recommended starting point, as is the "[Invitation to Julia](https://www.youtube.com/watch?v=gQ1y5NUD_RI)" workshop video from JuliaCon 2015  ([slide materials here](https://github.com/dpsanders/invitation_ ...
+[The learning page](https://julialang.org/learning/) has a great list of resources for new and experienced users alike. [This tutorial video](https://www.youtube.com/watch?v=vWkgEddb4-A) is one recommended starting point, as is the "[Invitation to Julia](https://www.youtube.com/watch?v=gQ1y5NUD_RI)" workshop video from JuliaCon 2015  ([slide materials here](https://github.com/dpsanders/invitation ...
```
Should the host name be changed to lowercase?

tkelman replied:
good catch, my sed and grep were case-sensitive so missed the later ones on this line

tkelman committed:
Fix unusually capitalized Julialang.org link

and remove separate link to docs search which doesn't exist with
new doc system

Sacha0 reviewed (approved):
lgtm (caveat: skimmed rapidly)!

StefanKarpinski commented:
Man, that touched a lot of files.

tkelman commented:
Yeah, license headers. Linkcheck doesn't touch the header comments so could split those out, the sed wasn't very discriminating.

StefanKarpinski commented:
No, it's totally fine – they should be updated too.
